payment/order.go

```diff
@@ -146,6 +155,91 @@ func CreateOrderItemFromMacaroon(sku string, quantity int) (*OrderItem, error) {
 	return &orderItem, nil
 }
 
+// IsStripePayable returns true if every item is payable by Stripe
+func (order Order) IsStripePayable() bool {
+	for _, item := range order.Items {
+
+		// check stripe in payment
+		if !strings.Contains(strings.Join(item.PaymentMethods, ","), StripePaymentMethod) {
+			return false
+		}
+		// TODO: make sure we have a stripe_product_id caveat
+		// TODO: if not we need to look into subscription trials:
+		/// -> https://stripe.com/docs/billing/subscriptions/trials
+	}
+	return true
+}
+
+// CreateCheckoutSessionResponse - the structure of a checkout session response
+type CreateCheckoutSessionResponse struct {
+	SessionID string `json:"checkoutSessionId"`
+}
+
+// CreateStripeCheckoutSession - Create a Stripe Checkout Session for an Order
+func (order Order) CreateStripeCheckoutSession(email, successURI, cancelURI string) (CreateCheckoutSessionResponse, error) {
+	// Create customer if not already created
+	i := customer.List(&stripe.CustomerListParams{
+		Email: stripe.String(email),
+	})
+
+	matchingCustomers := 0
+	for i.Next() {
+		matchingCustomers++
+	}
```

Comment:
I'm confused by this. Why isn't `matchingCustomers` just equal to the length of `customer.List`

Reply:
customer.List hands back an iterator which we are iterating through to get the coundt of matching customers.